runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/hotcold/HotColdSolver.java

```diff
@@ -51,7 +52,7 @@
 
 	public HotColdSolver(Set<HotColdLocation> possibleLocations)
 	{
-		this.possibleLocations = possibleLocations;
+		this.possibleLocations = new HashSet<>(possibleLocations);
```

Comment:
The `ImmutableSet`s from the solver test cause errors due to `removeIf()` in the solver. This could be reverted and the solver could use mutable sets instead if you prefer.

Reply:
For context, reverting this line causes this error in most of the tests:

```
java.lang.UnsupportedOperationException
	at com.google.common.collect.ImmutableCollection.removeIf(ImmutableCollection.java:273)
	at net.runelite.client.plugins.cluescrolls.clues.hotcold.HotColdSolver.signal(HotColdSolver.java:90)
	at net.runelite.client.plugins.cluescrolls.clues.hotcold.HotColdSolverTest.testSolver(HotColdSolverTest.java:234)
	at net.runelite.client.plugins.cluescrolls.clues.hotcold.HotColdSolverTest.testNarrowToFindSolutions(HotColdSolverTest.java:126)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	...
```

Reply:
@Adam- Any thoughts on this before I rebase this?